File: wms-technical-test-backend/internal/dto/response.go
package dto

type BaseResponse struct {
	Code    int         `json:"code"`
	Status  string      `json:"status"`
	Message string      `json:"message,omitempty"`
	Data    interface{} `json:"data,omitempty"`
}

type BaseResponseWithMessage struct {
	Code    int    `json:"code"`
	Status  string `json:"status"`
	Message string `json:"message"`
}

type BaseResponseWithData struct {
	Code   int         `json:"code"`
	Status string      `json:"status"`
	Data   interface{} `json:"data"`
}

type BaseResponseWithDataAndMeta struct {
	Code   int         `json:"code"`
	Status string      `json:"status"`
	Meta   interface{} `json:"meta"`
	Data   interface{} `json:"data"`
}

type BaseResponseWithMessageAndErrors struct {
	Code    int         `json:"code"`
	Status  string      `json:"status"`
	Message string      `json:"message"`
	Errors  interface{} `json:"errors"`
}

type BaseResponsePaginationMeta struct {
	Page       int   `json:"page"`
	PerPage    int   `json:"per_page"`
	Total      int64 `json:"total"`
	TotalPages int   `json:"total_pages"`
}

File: wms-technical-test-frontend/src/types/response.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
export type BaseResponse<TData = null | undefined> = {
  status?: string | null;
  message?: string | null;
  data?: TData | null;
};

export type PaginationBaseResponse<TData = null | undefined> = {
  status?: string | null;
  message?: string | null;
  data?: TData | null;
  meta?: {
    page?: number | null;
    limit?: number | null;
    total?: number | null;
    total_pages?: number | null;
  } | null;
};

export type BaseErrorResponse = {
  status?: string | null;
  message?: string | null;
  error?: {
    code?: string | null;
    details?: string | null;
    type?: string | null;
  } | null;
};

export type RejectWithValue = {
  statusCode?: number | null;
  message?: string | null;
  response?: any;
};
